Movies.web/Controllers/PaymentController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Movies.web.Models;
+
+
+namespace Movies.web.Controllers
+{
+    public class PaymentController : Controller
+    {
+        // GET: PaymentController
+        public ActionResult Index()
+        {
+
+
+            return View();
+        }
+
+        // GET: PaymentController/Details/5
+        public ActionResult Details(int id)
+        {
+            Payment payment = new Payment();
+            payment.CardNumber = Convert.ToInt64(Request.Form["CardNumber"]);
+            payment.OwnerName = Request.Form["Owner Name"].ToString();
+            payment.DateOfExpiry = Convert.ToInt32(Request.Form["Date of Expiry"]);
```

Comment:
Ya que recibes el Date of Expiry como string, podrías "jugar" con el string para obtener el año, mes y día y luego implementar algo como:

`new DateOnly(año, mes, dia)`

Reply:
Explícate, sabes que soy un noob

Reply:
@Ferlinmiguel por ejemplo, si esto `Request.Form["Date of Expiry"]` te da algo como `"26/04/2022"` puedes hacerle un split al string y partirlo donde sea que haya un `/` y ahí ya sabes que en la primera posición tienes el día, en la segunda el mes y en la tercera el año, entonces conviertes cada uno de esos 3 en Int y haces lo que te dije anteriormente.

Aquí puedes obtener un ejemplo de cómo hacer un split:
https://www.geeksforgeeks.org/string-split-method-in-c-sharp-with-examples/#:~:text=In%20C%23%2C%20Split()%20is,or%20an%20array%20of%20strings.